### PR TITLE
Select identifiable layers

### PR DESCRIPTION
**Pull request by gwenandres**
Hello, 

In this PR, I make the layer queryable icon clickable. User can choose which layers are queryable for Identify tool for example. 
I add an option in config.json to activate or not this feature.

image

Have a good day

_This feature has been funded by Grand Lyon Métropole https://www.grandlyon.com/_

### CODE REVIEW

**Comment by manisandro** [comment]
I'd do it like the visibility checkboxes, which can appear unchecked, tristate or checked.

File: utils/LayerUtils.js [hunk 1]
```diff
@@ -196,9 +197,48 @@ const LayerUtils = {
                 ...layer.dimensionValues
             };
         }
+
+        function findFirstDifferenceList(smallList, longList_) {
```
[the remark this answers is not in the document]
Comment (manisandro) [reply utils/LayerUtils.js:201]: Yes, but those are uses within `findFirstDifferenceList` itself?

[the remark this answers is not in the document]
Comment (gwenandres) [reply utils/LayerUtils.js:201]: This function ends at line 214.

I think it was a bad idea to put findFirstDifferenceList function in buildWMSLayerParams function. It is hard to read. 
I will write it outside :)

[the remark this answers is not in the document]
Comment (manisandro) [reply utils/LayerUtils.js]: I'm a bit concerned regarding initialQueryLayers vs queryLayers especially regarding db-permalinks. What about storing new a `omitFromQuerylayers` or similarly worded field in each sublayer, and just recomputing the toplevel `queryLayers` field in `buildWMSLayerParams`  resp `collectWMSSublayerParams` via

     sublayer.queryable && !sublayer.omitFromQuerylayers

I think this should also generally simplify the code somewhat, as simple call to `changeLayerProperty` for `omitFromQuerylayers` should suffice along with a simple change in `collectWMSSublayerParams`.

File: plugins/style/LayerTree.css [hunk 2]
```diff
@@ -181,6 +181,25 @@ img.layertree-item-legend-tooltip {
...
+#LayerTree span.layertree-item-identifyable-unchecked {
+    color: var(--border-color);
+}
+
+#LayerTree span.layertree-item-identifyable-checked {
+    color: var(--color-active);
+}
+
+#LayerTree span.layertree-item-identifyable-tristate {
+    color: var(--color---text-color);
```
Comment (manisandro) [inline plugins/style/LayerTree.css:200]: Small typo here?

File: utils/LayerUtils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gwenandres) [reply utils/LayerUtils.js]: Hello @manisandro thanks for your advices ! I think it is a better to deal with `omitFromQuerylayers`.

I am not sure if I have correctly used the sublayer attributes, especially during initialization... I didn't find where to initialize a new attribute but it is added with the use of `changeLayerProperty`.

**Comment by gwenandres** [comment]
image

**Comment by manisandro** [comment]
Thanks! Just a hint regarding the plugin documentation: you can write the docstring above the prop in the source code, and then run `yarn run plugindoc` from your qwc-app folder, which will then automatically generate the `qwc2/doc/plugins.md`.

**Comment by gwenandres** [comment]
Thanks !

**Pull request merged**